--- ed25519/src/seed.c	2013-12-28 20:00:28.526535356 -0500
+++ fixed-ed25519/seed.c	2013-12-28 19:05:43.402601524 -0500
@@ -29,12 +29,13 @@
     if (f == NULL) {
         return 1;
     }
-
-    fread(seed, 1, 32, f);
+    int items_read = fread(seed, 1, 32, f);
     fclose(f);
+    if (items_read != 32) {
+      return 1;
+    }
 #endif
-
     return 0;
 }
 
-#endif
\ No newline at end of file
+#endif
