--- ed25519/src/sha512.c	2013-12-28 20:00:28.526535356 -0500
+++ fixed-ed25519/sha512.c	2013-12-26 16:15:03.662288281 -0500
@@ -268,8 +268,8 @@
 {
     sha512_context ctx;
     int ret;
-    if (ret = sha512_init(&ctx)) return ret;
-    if (ret = sha512_update(&ctx, message, message_len)) return ret;
-    if (ret = sha512_final(&ctx, out)) return ret;
+    if ((ret = sha512_init(&ctx))) return ret;
+    if ((ret = sha512_update(&ctx, message, message_len))) return ret;
+    if ((ret = sha512_final(&ctx, out))) return ret;
     return 0;
-}
\ No newline at end of file
+}
